fem/fespace.hpp

```diff
@@ -587,6 +587,35 @@ class FiniteElementSpace
    bool Conforming() const { return mesh->Conforming() && cP == NULL; }
    bool Nonconforming() const { return mesh->Nonconforming() || cP != NULL; }
 
+   /** Set the prolongation operator of the space to an arbitrary sparse matrix,
+       taking ownership of the argument. */
```

Comment:
If `cP` is already defined, it does not take ownership.

Reply:
Thanks, I believe it actually makes a copy on both branches, so I updated the comment (and similar for `SetRestriction`).

Reply:
It looks like in the case `!cp`, it takes ownership by defining a new `unique_ptr`. In the other case, it makes a copy. This is confusing and may be causing a memory issue in the failing CI.

Reply:
I believe the `new SparseMatrix( *p )` in the `!cP` branch is doing a deep copy instead of creating a `unique_ptr` to the original argument's data, but I'll take a closer look at the failing CI soon and see if I can figure out what's going wrong.

Reply:
Right, that looks like a deep copy. I missed the `new SparseMatrix`.